swatch-product-configuration/src/main/java/com/redhat/swatch/configuration/registry/Variant.java

```diff
@@ -32,11 +34,25 @@
  * Variants all have the same billing model.
  */
 @Data
-@Builder
 public class Variant {
 
   @NotNull @NotEmpty private String tag; // required
-  private List<String> roles;
-  private List<String> engineeringIds;
-  private List<String> productNames;
+  private List<String> roles = new ArrayList<>();
+  private List<String> engineeringIds = new ArrayList<>();
+  private List<String> productNames = new ArrayList<>();
+
+  public Optional<Subscription> getSubscription() {
+    return SubscriptionConfigRegistry.getInstance().lookupSubscriptionByTag(tag);
+  }
+
+  public static Optional<Variant> findByRole(String role) {
+    return SubscriptionConfigRegistry.getInstance()
+        .lookupSubscriptionByRole(role)
+        .flatMap(
+            subscription ->
+                subscription.getVariants().stream()
+                    .filter(variant -> !variant.getRoles().isEmpty())
+                    .filter(variant -> variant.getRoles().contains(role))
```

Comment:
It will be better if you can combine two filters in one like below:
```
SubscriptionConfigRegistry.getInstance()
    .lookupSubscriptionByRole(role)
    .flatMap(subscription -> subscription.getVariants().stream()
        .filter(variant -> !variant.getRoles().isEmpty() && variant.getRoles().contains(role))
        .findFirst());
```